ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/PaladinCloudIntegrationForm.tsx

```diff
@@ -0,0 +1,199 @@
+import * as yup from 'yup';
+import React, { ReactElement } from 'react';
+import { Checkbox, Form, PageSection, TextInput } from '@patternfly/react-core';
+import usePageState from 'Containers/Integrations/hooks/usePageState';
+import FormMessage from 'Components/PatternFly/FormMessage';
+import FormLabelGroup from 'Containers/Integrations/IntegrationForm/FormLabelGroup';
+import FormSaveButton from 'Components/PatternFly/FormSaveButton';
+import FormCancelButton from 'Components/PatternFly/FormCancelButton';
+import { CloudSourceIntegration } from 'services/CloudSourceService';
+import IntegrationFormActions from '../IntegrationFormActions';
+import useIntegrationForm from '../useIntegrationForm';
+import { IntegrationFormProps } from '../integrationFormTypes';
+
+export const validationSchema = yup.object().shape({
+    cloudSource: yup.object().shape({
+        name: yup.string().trim().required('Integration name is required'),
+        type: yup.string().matches(/TYPE_PALADIN_CLOUD/),
+        credentials: yup.object().shape({
+            secret: yup
+                .string()
+                .test('secret-test', 'A token is required', (value, context: yup.TestContext) => {
+                    const requireSecretField =
+                        // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                        // @ts-ignore
+                        context?.from[2]?.value?.updateCredentials || false;
+
+                    if (!requireSecretField) {
+                        return true;
+                    }
+
+                    const trimmedValue = value?.trim();
+                    return !!trimmedValue;
+                }),
+        }),
+        paladinCloud: yup.object().shape({
+            endpoint: yup.string().required('Endpoint is required'),
+        }),
+        skipTestIntegration: yup.bool(),
+    }),
+    updatePassword: yup.bool(),
+});
+
+export type CloudSourceIntegrationFormValues = {
+    cloudSource: CloudSourceIntegration;
+    updateCredentials: boolean;
+};
+export const defaultValues: CloudSourceIntegrationFormValues = {
+    cloudSource: {
+        id: '',
+        name: '',
+        type: 'TYPE_PALADIN_CLOUD',
+        credentials: {
+            secret: '',
+        },
+        skipTestIntegration: true,
+        paladinCloud: {
+            endpoint: 'https://api.paladincloud.io',
+        },
+    },
+    updateCredentials: true,
+};
+
+function PaladinCloudIntegrationForm({
+    initialValues = null,
+    isEditable = false,
+}: IntegrationFormProps<CloudSourceIntegration>): ReactElement {
+    const formInitialValues = { ...defaultValues, ...initialValues };
+    if (initialValues) {
+        formInitialValues.cloudSource = { ...formInitialValues.cloudSource, ...initialValues };
```

Comment:
FYI again, not to change. When objects like `credentials` or `paladinCloud` have more than one property (these have only one) then lodash `merge` seems safer than object spread:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Integrations/IntegrationForm/Forms/JiraIntegrationForm.tsx#L132